SquirrelPanel.m

```diff
@@ -1143,7 +1144,7 @@ - (void)showPreedit:(NSString *)preedit
     if (i < comments.count && [comments[i] length] != 0) {
       [line appendAttributedString:[[NSAttributedString alloc]
                                        initWithString:@" "
-                                           attributes:attrs]];
+                                           attributes:commentAttrs]];
```

Comment:
This line is the needed change to fix vertical comment.

I cannot use your original comments as they are not atomic; a comment does more than one things and a fix is spread in several commits.

Reply:
typo in the comment: s/comment/commit/g

Reply:
Not only this, the key is to perform `convertToVerticalGlyph` right after each piece of line added. So I added a candidateStart int to record the start position of candidate. The same for a trailing label as well